### PR TITLE
Fix HMR support

### LINKED ISSUE #21: Storybook argtypes not being regenerated with HMR
# Related
https://github.com/storybookjs/storybook/issues/21729

### PR DESCRIPTION
**Pull request by joshwooding**
Closes #21

### CODE REVIEW

**Comment by bnussman-akamai** [comment]
Any update or is there any way we can help with this effort to fix HMR?

File: packages/vite-plugin-react-docgen-typescript/src/index.ts [hunk 1]
```diff
@@ -81,5 +101,8 @@ export default function reactDocgenTypescript(config: Options = {}): Plugin {
 				return src;
 			}
 		},
+		async handleHotUpdate() {
```
Comment (joshwooding) [inline packages/vite-plugin-react-docgen-typescript/src/index.ts:104]: The magic sauce

Comment (bnussman-akamai) [reply packages/vite-plugin-react-docgen-typescript/src/index.ts:104]: YAYA 🎉

File: packages/vite-plugin-react-docgen-typescript/src/index.ts [hunk 2]
```diff
@@ -30,38 +60,38 @@ const getUtils = async (config: Options) => {
...
-	const result = {
-		docGenParser,
-		filter,
-		generateOptions,
-		generateDocgenCodeBlock,
-		tsProgram,
-	};
-
-	return result;
+	return ts.createProgram(files, compilerOptions, undefined, oldProgram);
```
Comment (valentinpalkovic) [inline packages/vite-plugin-react-docgen-typescript/src/index.ts:63]: Hey @joshwooding 

What happens if we pass the `oldProgram` to the `ts.createProgram` function? Is Typescript able to speed up the new program's creation?

Comment (joshwooding) [reply packages/vite-plugin-react-docgen-typescript/src/index.ts:63]: Yeah, that's in theory how it works

**Pull request merged**